src/devices/machine/mm5740.h

```diff
@@ -0,0 +1,210 @@
+				 the control character mode.
+
+Shift Lock I/O       20          This pin is intended to serve as an
+				 input when the shift lock key is
+				 depressed. It places the encoder
+				 in the shift mode. Upon release of
+				 the key, the shift mode will be
+				 maintained and this pin will serve
+				 as an output to drive an indicator.
+				 This function is reset by depressing
+				 the shift key.
+
+Clock                3           A TTL compatible clock signal is
+				 applied to this pin. A bit time is
+				 defined as the time from one
+				 negative going transition to the
+				 succeeding negative going transition
+				 of the clock.
+
+Vss                  32          +5.0V supply
+
+Vll                  2           Ground
+
+Vgg                  18          -12V supply
+
+
+**********************************************************************/
+
+#pragma once
+
+#ifndef __MM5740__
+#define __MM5740__
+
+#include "emu.h"
+
+
+
+//**************************************************************************
+//  INTERFACE CONFIGURATION MACROS
+//**************************************************************************
+
+#define MCFG_MM5740_MATRIX_X1(_cb)       devcb = &mm5740_device::set_x1_cb(*device, DEVCB_##_cb);
+#define MCFG_MM5740_MATRIX_X2(_cb)       devcb = &mm5740_device::set_x2_cb(*device, DEVCB_##_cb);
+#define MCFG_MM5740_MATRIX_X3(_cb)       devcb = &mm5740_device::set_x3_cb(*device, DEVCB_##_cb);
+#define MCFG_MM5740_MATRIX_X4(_cb)       devcb = &mm5740_device::set_x4_cb(*device, DEVCB_##_cb);
+#define MCFG_MM5740_MATRIX_X5(_cb)       devcb = &mm5740_device::set_x5_cb(*device, DEVCB_##_cb);
+#define MCFG_MM5740_MATRIX_X6(_cb)       devcb = &mm5740_device::set_x6_cb(*device, DEVCB_##_cb);
+#define MCFG_MM5740_MATRIX_X7(_cb)       devcb = &mm5740_device::set_x7_cb(*device, DEVCB_##_cb);
+#define MCFG_MM5740_MATRIX_X8(_cb)       devcb = &mm5740_device::set_x8_cb(*device, DEVCB_##_cb);
+#define MCFG_MM5740_MATRIX_X9(_cb)       devcb = &mm5740_device::set_x9_cb(*device, DEVCB_##_cb);
+#define MCFG_MM5740_SHIFT_CB(_cb)        devcb = &mm5740_device::set_shift_cb(*device, DEVCB_##_cb);
+#define MCFG_MM5740_CONTROL_CB(_cb)      devcb = &mm5740_device::set_control_cb(*device, DEVCB_##_cb);
+#define MCFG_MM5740_DATA_READY_CB(_cb)   devcb = &mm5740_device::set_data_ready_cb(*device, DEVCB_##_cb);
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+// ======================> mm5740_device
+
+class mm5740_device :   public device_t
+{
+public:
+	// construction/destruction
+	mm5740_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
+	// public interface
+	uint16_t b_r();
+
+	template<class _Object> static devcb_base &set_x1_cb(device_t &device, _Object rd) { return downcast<mm5740_device &>(device).m_read_x1.set_callback(rd); }
```

Comment:
Please make these take the `_Object` as a universal reference and forward it to allow for callbacks that are movable but not copyable.